var/spack/repos/builtin/packages/vtkm/package.py

```diff
@@ -17,39 +17,77 @@ class Vtkm(Package):
+    variant("logging", default=False, description="build logging support")
+    variant("mpi", default=False, description="build mpi support")
+    variant("openmp", default=False, description="build openmp support")
+    variant("rendering", default=True, description="build rendering support")
     variant("tbb", default=True, description="build TBB support")
 
     depends_on("cmake")
     depends_on("tbb", when="+tbb")
     depends_on("cuda", when="+cuda")
+    depends_on("mpi", when="+mpi")
 
     def install(self, spec, prefix):
         with working_dir('spack-build', create=True):
             cmake_args = ["../",
                           "-DVTKm_ENABLE_TESTING=OFF",
-                          "-DVTKm_BUILD_RENDERING=ON",
                           "-DVTKm_USE_64BIT_IDS=OFF",
                           "-DVTKm_USE_DOUBLE_PRECISION=ON"]
-            # tbb support
-            if "+tbb" in spec:
-                # vtk-m detectes tbb via TBB_ROOT env var
-                os.environ["TBB_ROOT"] = spec["tbb"].prefix
-                cmake_args.append("-DVTKm_ENABLE_TBB=ON")
 
             # cuda support
             if "+cuda" in spec:
                 cmake_args.append("-DVTKm_ENABLE_CUDA=ON")
-                # this fix is necessary if compiling platform has cuda, but
-                # no devices (this common for front end nodes on hpc clusters)
-                # we choose kepler as a lowest common denominator
-                cmake_args.append("-DVTKm_CUDA_Architecture=kepler")
+                if 'cuda_arch' in spec.variants:
+                    cuda_arch = spec.variants['cuda_arch'].value
+                    cmake_args.append(
+                        '-DVTKm_CUDA_Architecture={0}'.format(cuda_arch[0]))
+                else:
+                    # this fix is necessary if compiling platform has cuda, but
+                    # no devices (this's common for front end nodes on hpc clus
+                    # ters)
+                    # we choose kepler as a lowest common denominator
+                    cmake_args.append("-DVTKm_CUDA_Architecture=kepler")
+
+            # logging support
+            if "+logging" in spec:
+                cmake_args.append("-DVTKm_ENABLE_LOGGING:BOOL=ON")
+
+            # mpi support
+            if "+mpi" in spec:
+                cmake_args.append("-DVTKm_ENABLE_MPI:BOOL=ON")
+
+            # openmp support
+            if "+openmp" in spec:
+                # openmp is added after version 1.3.0
+                if self.spec.satisfies('@:1.2.0') and \
+                        spec['vtkm'].version.string != 'master':
+                    raise ValueError('OpenMP is not supported for\
+                            vtkm version lower than 1.3')
+                # vtkm requires openmp no less than 4.0.0
+                if spec.compiler.satisfies('gcc@:4.8.9'):
+                    raise ValueError('The reqired OpenMP 4.0.0 is only\
+                            supported with gcc no less than 4.9')
```

Comment:
It would be better to change both of these to `conflicts`. That way, things will crash during concretization instead of later during installation.